keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1144 @@
+
+If a cluster is upgraded, the cluster administrator will experience pre-upgrade behavior until 
+`resources` field is specified at pod-level. 
+
+If the workloads are not using the new feature, downgrade won't be affected. However, if it is required to
+downgrade a cluster that has workloads using the pod-level limits functionality, following steps need to be
+performed:
+
+* Translate pod-level resource specifications to container-level specifications.
+
+* Delete existing workloads and recreate them with container-level specifications.
+
+* Drain the nodes one by one to prevent the new pods from being scheduled on them.
+
+* Downgrade the kubelet on the drained nodes to a version that doesn't support pod-level resource specification
+fields.
+
+* Downgrade the control plane components.
+
+* Uncordon the nodes, making it available for scheduling again.
+
+
+### Version Skew Strategy
+
+Pod-level resource specification is an opt-in feature. For this feature to work correctly, it must be enabled
+in all parts of the cluster (scheduler, API server, kubelet). However, if a node is running an older version
+of kubelet that doesn't support this feature, it will ignore the pod-level resouce specifications and continue
+to use the aggregated container-level values (backward compatible). Applications must be prepared for this version
+skew implementation.
+
+For users wanting to use this feature, it is user's responsibility to use the correct version of kube-scheduler,
+kube-apiserver and kubelet otherwise the feature will not work as expected.
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: PodLevelResources
+  - Components depending on the feature gate: kubelet, kube-apiserver, kube-scheduler
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No. `resources` must be set/unset in `spec` at pod-level to enable/disable this feature.
+    The workloads need to restart to disable the feature.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No. Only the workloads need to restart.
+
+###### Does enabling the feature change any default behavior?
+
+Yes, if the feature is enabled and pod level resources are specified the scheduler and kubelet will consider
+pod level values over container level resource specs. 
```

Comment:
I think this doesn't count as default behavior, since the user needs to set a new field for it to work. If the user doesn't set pod level resources, there is no change to behavior, right?